### PR TITLE
Add Qt5/QML building framework to contrib/android:

### PR DESCRIPTION
**Pull request by ecdsa**
- update requirements file for building PyQt5
 - use separate .buildozer directories for qml and Kivy
 - add GUI and ARCH parameters to android/build.sh

@SomberNight can you have a look? I would like to have this merged ahead of the `android-qml` branch, because `contrib/android` gets modified frequently.

Maybe we should host our own `python-for-android` fork in `spesmilo`. I am not sure about the semantics of the change to `PyjniusRecipe.depends`, but it seems that adding qt5 there does not impact the kivy installation.

@accumulator I have un-commented the `make theming` line in the Makefile, I don't think it will impact your builds

### CODE REVIEW

**Comment by SomberNight** [comment]
>> `assert PyjniusRecipe.depends == [('genericndkbuild', 'sdl2', 'qt5'), 'six', 'python3']`
>
> I am not sure about the semantics of the change to PyjniusRecipe.depends, but it seems that adding qt5 there does not impact the kivy installation.

The round brackets `()` stand for logical `OR`, while the square brackets `[]` stand for logical `AND`, so the change looks fine.

**Comment by accumulator** [comment]
> @accumulator I have un-commented the `make theming` line in the Makefile, I don't think it will impact your builds

That's ok

File: contrib/deterministic-build/requirements-build-android.txt [hunk 1]
```diff
@@ -164,4 +164,22 @@ zipp==3.7.0 \
...
+    --hash=sha256:87539b534d730aaabde7396bd2edb7c5b86093ca2500eeaa617d588150b7fda9
+pyparsing==3.0.6 \
+    --hash=sha256:04ff808a5b90911829c55c4e26f75fa5ca8a2f5f36aa3a51f68e27033341d3e4 \
+    --hash=sha256:d9bdec0013ef1eb5a84ab39a3b3868911598afa494f5faa038647101504e2b81
+PyQt-builder==1.12.2 \
+    --hash=sha256:48f754394d235307201ec2b5355934858741201af09433ff543ca40ae57b7865 \
+    --hash=sha256:f62bb688d70e0afd88c413a8d994bda824e6cebd12b612902d1945c5a67edcd7
+packaging==21.3 \
+    --hash=sha256:dd47c42927d89ab911e606518907cc2d3a1f38bbd026385970643f9c5b8ecfeb \
+    --hash=sha256:ef103e05f519cdc783ae24ea4e2e0f508a9c99b2d4969652eed6a2e1ea5bd522
```
Comment (SomberNight) [inline contrib/deterministic-build/requirements-build-android.txt:185]: I haven't noticed that `contrib/requirements/requirements-build-android.txt` was not modified.
`contrib/deterministic-build/requirements*` is typically generated from `contrib/requirements/requirements*`, using the `contrib/freeze_packages.sh` script.
i.e. the file here is a generated file.
Are there any specific version requirements, or do we just require `PyQt-builder` and `typing-extensions`, (the former pulling in the rest)?

Comment (SomberNight) [reply contrib/deterministic-build/requirements-build-android.txt:185]: did you generate this diff (the hashes) manually?
see if https://github.com/spesmilo/electrum/pull/7712 looks ok

Comment (accumulator) [reply contrib/deterministic-build/requirements-build-android.txt:185]: I think I know where this comes from. When I started the qt5 bootstrap for p4a I updated only the `contrib/deterministic-build/requirements-build-android.txt `in my branch, and those patches got merged to master recently as part of the effort to parameterize the build to kivy and qml options.

`PyQt-builder` and `typing-extensions` versions are just latest, no specific version required.

**Pull request merged**